Cookie/CookieChecker.php

```diff
@@ -20,9 +21,9 @@ class CookieChecker
      */
     private $request;
 
-    public function __construct(Request $request)
+    public function __construct(RequestStack $request)
     {
-        $this->request = $request;
+        $this->request = $request->getCurrentRequest();
```

Comment:
You shouldn't call `getCurrentRequest()` in the constructor, but in the method in which it is used. The reason is, that the constructor is called during service generation (when it is injected into another service) but the actual usage may be later, so the "current request" may be another one at that point.

https://symfony.com/doc/current/service_container/request.html